x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_diagnostics/index.tsx

```diff
@@ -235,12 +254,21 @@ export const AgentDiagnosticsTab: React.FunctionComponent<AgentDiagnosticsProps>
         </EuiCallOut>
       </EuiFlexItem>
       <FlexStartEuiFlexItem>
-        <EuiButton fill size="m" onClick={onSubmit} disabled={isSubmitting}>
-          <FormattedMessage
-            id="xpack.fleet.agentList.diagnosticsOneButton"
-            defaultMessage="Request diagnostics .zip"
-          />
-        </EuiButton>
+        {isAgentRequestDiagnosticsSupported(agent) ? (
+          requestDiagnosticsButton
+        ) : (
+          <EuiToolTip
+            content={
+              <FormattedMessage
+                id="xpack.fleet.requestDiagnostics.notSupportedTooltip"
+                defaultMessage="Request diagnostics action is not supported for agents before version {version}."
```

Comment:
"action" is sort of an implementation detail, so I think we can improve the wording here

```suggestion
                defaultMessage="Requesting agent diagnostics is not supported for agents before version {version}."
```